scaler/object_storage/object_storage_server.cpp

```diff
@@ -104,72 +118,103 @@ void ObjectStorageServer::closeServerReadyFds()
     }
 }
 
-awaitable<void> ObjectStorageServer::listener(tcp::endpoint endpoint)
+ObjectRequestHeader ObjectStorageServer::parseRequestHeader(Bytes message)
 {
-    auto executor = co_await boost::asio::this_coro::executor;
-    tcp::acceptor acceptor(executor, endpoint);
+    assert(message.len() == CAPNP_HEADER_SIZE);
+    std::array<uint64_t, ObjectRequestHeader::bufferSize() / CAPNP_WORD_SIZE> buffer {};
+    memcpy(buffer.data(), message.data(), message.size());
+    return ObjectRequestHeader::fromBuffer(buffer);
+}
 
-    _logger.log(scaler::ymq::Logger::LoggingLevel::info, "ObjectStorageServer: started");
+void ObjectStorageServer::processRequests()
+{
+    using namespace std::chrono_literals;
+    try {
+        std::map<ymq::Configuration::IOSocketIdentity, std::pair<ObjectRequestHeader, Bytes>> identityToFullRequest;
+        std::vector<std::future<std::expected<void, ymq::Error>>> pendingSendMessageFuts;
 
-    setServerReadyFd();
+        while (true) {
+            auto invalids = std::ranges::partition(pendingSendMessageFuts, [](const auto& x) { return x.valid(); });
+            pendingSendMessageFuts.resize(std::distance(pendingSendMessageFuts.begin(), invalids.begin()));
```

Comment:
Isn't this `std::ranges::remove_if`?

```suggestion
            std::ranges::remove_if(pendingSendMessageFuts, [](const auto& x) { return !x.valid(); });
```